### DISCUSSION TITLE
How can you combine both prompt functions and native code functions in one plugin?

### DISCUSSION THREAD

**Discussion by Valkozaur** [microsoft/semantic-kernel#5709] [Q&A] [answered]
Hello SK users. 

I am developing a chat application. 
While developing I started wondering what are the options to use both native code functions and prompt functions (yml or skprompt.txt   + config.json) in one plugin?

Is that possible?

**Comment by stephentoub** [comment, marked as answer]
You can create a new plugin that contains all the functions from other plugins, e.g.
```C#
KernelPlugin plugin1 = ...;
KernelPlugin plugin2 = ...;
KernelPlugin combined = KernelPluginFactory.CreateFromFunctions("MyCoolPlugin", [..plugin1, ..plugin2]);
```
You can also write your own KernelPlugin-derived types. Here's an example of writing and using a mutable one that let's you add and remove functions.
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example69_MutableKernelPlugin.cs

**Reply by Valkozaur** [reply]
Thanks @stephentoub. 
BTW great video for the async and await with Scott.
